### PR TITLE
Add examples to the docs page

### PR DESCRIPTION
**Pull request by mforets**
This PR brings the examples script (see [here](https://github.com/JuliaIntervals/TaylorModels.jl/pull/95)) to the main docs page.

I may contribute w/other subsections to that page (example use cases that come up frequently on the issues list of this repo / on Discourse), but we can go in stages.. :)

### CODE REVIEW

**Comment by mforets** [comment]
This PR is only an update to the docs. As you may know, Documenter has a feature to push a preview of the docs if you set `push_preview=true` in the call to `deploydocs`. Otherwise, you can pull this forked branch to build the docs locally with:

```
$ git fetch origin pull/98/head:pr/98 && git checkout pr/98
```
I always forget that trick :)

**Review by lbenet — COMMENTED** [review]
Thanks a lot for your contribution, @mforets! 

I left a comment about the wording, which sounds better to me, but be aware that I'm not a native English speaker...

In addition, I'd like to suggest to use `push_preview = true` within the `deploydocs` function (in `docs/make.jl`). It allows to have a preview (online) of the docs; see [this](https://juliadocs.github.io/Documenter.jl/stable/lib/public/#Documenter.deploydocs) for the  relevant documentation. (This may make life easier for #99.)

File: docs/src/index.md [hunk 1]
```diff
@@ -33,6 +34,176 @@ plot!(ftm7, label="7th order")
...
+
+```@example range_bound_1d
+Dm = mince(D, 8) # split the domain into 8 smaller chunks
+
+Em = evaluate.(s, Dm) # evaluate the Taylor model on each sub-domain
+
+Rm = reduce(hull, Em) # take the convex hull, i.e. the smallest interval that contains them all
+```
+
+Here the lower bound has improved by mincing (or splitting) the domain, and it
```
Comment (lbenet) [inline docs/src/index.md:113]: "has  improved" -> "has been improved"

(Sorry, forgot to click the button.)

Comment (mforets) [inline docs/src/index.md:113]:
```suggestion
Here the lower bound has been improved by mincing (or splitting) the domain, and it
```

File: docs/src/index.md [hunk 2]
```diff
@@ -33,6 +34,176 @@ plot!(ftm7, label="7th order")
...
+
+```@example range_bound_1d
+domain(t)
+```
+
+```@example range_bound_1d
+expansion_point(t)
+```
+
+Finally, note that the Taylor model type is a parametric structure with two fields,
```
Comment (mforets) [inline docs/src/index.md:178]: here `field` is wrong, the correct way to say is probably "is a structure with two parameters ..."

Comment (lbenet) [reply docs/src/index.md:178]: What about "types"?

**Comment by mforets** [comment]
@lbenet  I've updated this branch with master, however i'm not sure if docs preview will work because the PR is made from my fork (I don't have rights to make it from JuliaIntervals).

**Comment by lbenet** [comment]
Thanks a lot; I'll take a look on them later... 

I think you need to rebase to master (and push again), since #100 has been merged.  this should make possible to see the preview of  the  docs.

**Comment by mforets** [comment]
> I think you need to rebase to master (and push again), since #100 has been merged. this should make possible to see the preview of the docs.

yes, i did just that. about my comment on the fork is about [this Note](https://juliadocs.github.io/Documenter.jl/stable/lib/public/#Documenter.deploydocs):

> Pull requests made from forks will not have previews. Hosting previews requires access to the deploy key. Therefore, previews are available only for pull requests that were submitted directly from the main repository.

**Comment by lbenet** [comment]
Mmmm... you are right. Since the proposed changes LGTM, I guess  we can simply go  ahead and merge them. Is this ready?

**Comment by mforets** [comment]
from my side it is ready 👌

**Review by lbenet — APPROVED** [review]

**Comment by lbenet** [comment]
Thanks  a  lot @mforets!! Merging.

**Pull request merged**